src/karma-webpack.js

```diff
@@ -91,23 +100,56 @@ function Plugin(
   var applyPlugins = compiler.compilers || [compiler]
 
   applyPlugins.forEach(function(compiler) {
-    compiler.plugin('this-compilation', function(compilation, params) {
-      compilation.dependencyFactories.set(SingleEntryDependency, params.normalModuleFactory)
-    })
-    compiler.plugin('make', this.make.bind(this))
-  }, this);
+    if (compiler.hooks) {
+      compiler.hooks
+        .thisCompilation
+        .tap(this.plugin, (compilation, params) => {
+          compilation.dependencyFactories.set(SingleEntryDependency, params.normalModuleFactory)
+        })
+      compiler.hooks
+        .make
+        .tapAsync(this.plugin, this.make.bind(this))
+    } else {
+      compiler.plugin('this-compilation', function(compilation, params) {
+        compilation.dependencyFactories.set(SingleEntryDependency, params.normalModuleFactory)
+      })
+      compiler.plugin('make', this.make.bind(this))
+    }
+  }, this)
+
+  function handler(callback) {
+    isBlocked = true
+
+    if (typeof callback === 'function') {
+      callback(null)
+    }
+  }
 
-  ['invalid', 'watch-run', 'run'].forEach(function(name) {
-    compiler.plugin(name, function(_, callback) {
-      isBlocked = true
+  var hooks = ['invalid', 'watch-run', 'run']
 
-      if (typeof callback === 'function') {
-        callback()
+  if (compiler.hooks) {
+    hooks = [
+      {method: 'sync', name: 'invalid'},
+      {method: 'async', name: 'watchRun'},
+      {method: 'async', name: 'run'}
+    ]
+  }
+
+  hooks.forEach(function(hook) {
```

Comment:
`function` is needed here to keep the context ? :)

Reply:
I think yes, as there is the second argument in line 150 which set `this` for the callback. The plugin uses ES5 syntax, not ES6, so I think it is ok to use the function here.